gcloud-java-compute/src/main/java/com/google/gcloud/spi/ComputeRpc.java

```diff
@@ -293,9 +294,44 @@ public Y y() {
   /**
    * Deletes the requested region address.
    *
-   * @return a region operation if request was issued correctly, {@code null} if the address was not
-   *     found
+   * @return a region operation if the request was issued correctly, {@code null} if the address was
+   *     not found
    * @throws ComputeException upon failure or if region is not found
    */
   Operation deleteRegionAddress(String region, String address, Map<Option, ?> options);
+
+  /**
+   * Creates a snapshot for the specified disk.
+   *
+   * @return a zone operation if the create request was issued correctly, {@code null} if the disk
+   *     was not found
+   * @throws ComputeException upon failure
+   */
+  Operation createSnapshot(String zone, String disk, String snapshot, String description,
```

Comment:
My idea is that as soon as we have all identities implemented the Operation class would become:

`Operation<T extends ResourceId>`

Where the `operation.targetId` field is of type `T`.
So that each method in compute returns the proper `Operation<ResourceId>`.
With this in mind I think making a distinction between region/zone/global operation is no longer needed. What do you think? 

Anyway I prefer not to address this in the current PR, this also deserves a dedicated issue that I will open with the content of this discussion.